Tests/Marketplace/search_and_install_packs.py

```diff
@@ -319,86 +336,71 @@ def install_packs(client: demisto_client,
-            return install_packs(client, host, [pack for pack in packs_to_install if pack['id'] not in e.malformed_ids],
-                                 request_timeout)
-
-        except GCPTimeOutException as e:
-            # if this is a gcp timeout, try only once more
-            logging.warning(f'The request to install packs on server {host} has failed due to timeout awaiting response'
-                            f' headers while trying to install pack {e.pack_id}, trying again for one more time')
-            call_install_packs_request(packs_to_install)
-
-        except GeneralItemNotFoundError as e:
-            logging.warning(f'The request to install all packs on server {host} has failed due to an item not found '
-                            f'error, with the message: {e.error_msg}.\n trying again for one more time')
-            call_install_packs_request(packs_to_install)
+                if malformed_ids := find_malformed_pack_id(ex.body):
+                    handle_malformed_pack_ids(malformed_ids, packs_to_install)
+                    if not attempt:
+                        raise Exception(f"malformed packs:{malformed_ids}") from ex
 
+                    # We've more attempts, retrying without tho malformed packs.
+                    SUCCESS_FLAG = False
+                    logging.error(f"Unable to install malformed packs:{malformed_ids}, retrying without them.")
+                    packs_to_install = [pack for pack in packs_to_install if pack['id'] not in malformed_ids]
+
+                if (error_ids := get_error_ids(ex.body)) and TEST_MODELING_RULE_ERROR_CODE in error_ids:
+                    # If we got this error code, it means that the modeling rules are not valid, exiting install flow.
+                    raise Exception("Test modeling rules for pack are not valid") from ex
+
+                if not attempt:  # exhausted all attempts, understand what happened and exit.
+                    if 'timeout awaiting response' in ex.body:
+                        if '/packs/' in ex.body:
+                            pack_id = get_pack_id_from_error_with_gcp_path(ex.body)
+                            raise Exception(f"timeout awaiting response headers while trying to install pack {pack_id}") from ex
+
+                        raise Exception("timeout awaiting response headers while trying to install, "
+                                        "couldn't determine pack id.") from ex
+
+                    if 'Item not found' in ex.body:
+                        raise Exception(f'Item not found error, headers:{ex.headers}.') from ex
+
+                    # Unknown exception reason, re-raise.
+                    raise Exception(f"Got {ex.status} from server, message:{ex.body}, headers:{ex.headers}") from ex
+            except (HTTPError, HTTPWarning) as http_ex:
+                if not attempt:
+                    raise Exception("Failed to perform http request to the server") from http_ex
```

Comment:
```suggestion
                    raise Exception("Failed to perform http request to the server to install packs {packs_to_install}") from http_ex
```

Reply:
ignoring, the pack ids are printed in the outer exception block